pkg/kubelet/config/flags.go

```diff
@@ -95,15 +95,23 @@ func (s *ContainerRuntimeOptions) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.DockershimRootDirectory, "experimental-dockershim-root-directory", s.DockershimRootDirectory, "Path to the dockershim root directory.")
 	fs.MarkHidden("experimental-dockershim-root-directory")
 	fs.StringVar(&s.PodSandboxImage, "pod-infra-container-image", s.PodSandboxImage, fmt.Sprintf("The image whose network/ipc namespaces containers in each pod will use. %s", dockerOnlyWarning))
+	fs.MarkDeprecated("pod-infra-container-image", "will be removed along with dockershim.")
```

Comment:
if we remove this one with docker-shim, We need some way to get information about sandbox image from container runtime https://github.com/kubernetes/kubernetes/issues/81756.

Reply:
@adisky indeed. let's track that as a new issue, we may prefer another flag as this one started with docker and then was co-opted by gc manager if i remember correctly.

Reply:
https://github.com/kubernetes/kubernetes/issues/98765

Reply:
+1 on re-using the flag because of GC.

eventually CRI implementers should dump their info properly the way containerd is doing today on it's socket - you can try `crictl info` for that.
but currently not e.g. cri-o and dockershim.

Reply:
Removed the deprecation from this flag. improved the description as well.